### PR TITLE
Update links to WCAG criteria to make them clearer

### PR DESCRIPTION
**Pull request by calvin-lau-sig7**
## What

Changes identified unclear links on all WCAG-related links on the Design System.

This was done as a batch job to address about 40 flagged links in the link text audit, but these changes have not been drafted or approved there.

Full review of the changes in this PR is needed before publish.

### What we've done

All instances of link text to WCAG 2.2 criteria are clearly identified as WCAG 2.2 criteria, with SC number and name

On the top-of-page WAG 2.2 call-outs, we've removed "users can successfully" at the end of the lead-in line into the anchor links. 

> From:
>To [use component/pattern] and meet the new Web Content Accessibility Guidelines (WCAG) 2.2 criteria, make sure that users can successfully:
>
>To:
>To [use component/pattern] and meet the new Web Content Accessibility Guidelines (WCAG) 2.2 criteria, make sure that:

As a result, we've edited all the anchor links to be more descriptive within the link text, instead of relying on the lead-in sentence.

## Why

This more clearly communicates what the link is and where it goes to. Part of https://github.com/alphagov/govuk-design-system/issues/4267.

### CODE REVIEW

**Comment by calvin-lau-sig7** [comment]
> Question for you @calvin-lau-sig7: Do you think that instances of components need a 'component' suffix? For example the second callout link for back links is now 'users can interact with Back links' and I wonder if it should be 'users can interact with Back link _components_'. I'm not 100% about that as feedback because the user has the context of the page they're on so interested in your thoughts here.

Yeah, I better add the "component/pattern" words in... 
I've got a distinction in my head between the component itself, and the page control – like the "Back links component" is the code that generates individual "back links" throughout a service. 

But seeing as we're taking the wordy approach as standard, it makes sense for us to go use the wordy version everywhere it makes sense to.

I'll edit the PR...

**Comment by calvin-lau-sig7** [comment]
@selfthinker As mentioned in standup, here's a few things I wanted to mention for your awareness.

I'm looking to make these widespread edits:
- all links to WCAG 2.2 criteria now take a long form, formatted as "WCAG 2.2 success criterion 3.3.7 Redundant entry" or "WCAG 2.2 success criterion 3.3.7 Redundant entry, level AA"
- instances of WCAG 2.1 changed to WCAG 2.2

The style format for the success criterion is my best interpretation of following the GOV.UK style guide – alas WCAG's style for success criterion is inconsistent across their website.

I'm open to any other comments you have on these changes as well!

**Comment by calvin-lau-sig7** [comment]
>IMO ideally the callout links would also include the 'make sure that' from the lead-in paragraph to make it as clear as possible but I think that might tip this over into make the link too wordy and the experience bad for everyone else.

Looking at this again, this might be a good move... let me have a think 🤔

**Comment by calvin-lau-sig7** [comment]
I've taken in @owenatgov's suggestion to move the "make sure that" wording into the links, as I've agreed that it'll make the link text much clearer as messages. I've made the edits as a new commit and will verify it all looks good in preview tomorrow.

We'll need to review these in PR before push. cc @selfthinker

File: src/components/button/index.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ Do not only rely on the red colour of a warning button to communicate the seriou
 
 Use the `govuk-button--inverse` modifier class to show white buttons on dark backgrounds – for example, in headers, custom components, and patterns with darker backgrounds.
 
-Make sure all users can see the button – the white button and background colour [must have a contrast ratio of at least 3:1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html).
+Make sure all users can see the button – the white button and background colour must have a contrast ratio of at least 4.5:1 with white to meet [WCAG 2.2 success criterion 1.4.3 Contrast (minimum), level AA](https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html).
```
Comment (selfthinker) [inline src/components/button/index.md:101]: If you want to keep this style, you'd have to remove "white button and". Otherwise the white would be repeated.

File: src/get-started/focus-states/index.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ description: Some people use keyboards or other devices to navigate through a pa
 
 Some people use keyboards or other devices to navigate through a page by jumping from one interactive element to the next. Focus states let users know which element they’re currently on and is ready to be interacted with.
 
-Focus states in the GOV.UK Design System use a combination of yellow and black to make sure they meet [Web Content Accessibility Guidelines (WCAG) 2.2 success criterion 1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) on any background colour used on GOV.UK.
+Focus states in the GOV.UK Design System use a combination of yellow and black to make sure they meet [Web Content Accessibility Guidelines (WCAG) 2.2 success criterion 1.4.11 Non-text contrast, level AA](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html) on any background colour used on GOV.UK.
```
Comment (selfthinker) [inline src/get-started/focus-states/index.md:12]: Technically, the combination of yellow and black is not accessible at all. What makes the focus states accessible for 1.4.11 is the thick bottom border.

Rewriting that would be out of scope, though, because this PR doesn't change anything about that text.

File: src/patterns/create-accounts/index.md [hunk 3]
```diff
@@ -19,11 +19,11 @@ Help users create an account for your service.
   name: "Create accounts",
   criteria: [
     {
-      text: "reuse information they've already entered",
+      text: "make sure users can reuse information they've already entered",
       anchor: "wcag-reuse-account-information"
     },
     {
-      text: "avoid doing a cognitive test, such as a CAPTCHA, to create an account",
+      text: "make sure that cognitives tests, such as CAPTCHA, are not needed for users to create an account",
```
Comment (selfthinker) [inline src/patterns/create-accounts/index.md:26]: Did you remove the 'a' intentionally? If yes, the 'CAPTCHA' should end in an 's'.

File: src/components/breadcrumbs/index.md [hunk 4]
```diff
@@ -45,7 +45,7 @@ The breadcrumbs should start with your 'home' page and end with the parent secti
 
 {% call wcagNote({id: "wcag-interact-breadcrumbs"}) %}
 
-<p>Make sure no other interactive elements are too close to the breadcrumbs. This is to make sure users can easily interact with the breadcrumbs. This relates to WCAG 2.2 success criterion <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">2.5.8 Target size (minimum)</a>.</p>
+<p>Make sure no other interactive elements are too close to the Breadcrumbs component. This is to make sure users can easily interact with the Breadcrumbs component. This relates to <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```
[the remark this answers is not in the document]
Comment (calvin-lau-sig7) [reply src/components/breadcrumbs/index.md:48]: Yeah, the repetition was originally put in to avoid ambiguity, but I'm agreeing with your point. I'll probably change these...

File: src/patterns/create-accounts/index.md (hunk 3 above)
Comment (calvin-lau-sig7) [reply src/patterns/create-accounts/index.md:26]:
```suggestion
      text: "make sure that cognitives tests, such as CAPTCHAs, are not needed for users to create an account",
```

File: src/components/button/index.md [hunk 5]
```diff
@@ -50,10 +50,10 @@ Align the primary action button to the left edge of your form.
 
 {% call wcagNote({id: "wcag-interact-button"}) %}
 
-<p>Do not decrease the height or target area of buttons. This is to make sure users can easily interact with buttons. This is to comply with WCAG 2.2 success criterion <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">2.5.8 Target size (minimum)</a>.</p>
+<p>Do not decrease the height or target area of the Button component. This is to make sure users can easily interact with the Button component. This is to comply with <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```
Comment (calvin-lau-sig7) [inline src/components/button/index.md:53]:
```suggestion
<p>Do not decrease the height or target area of the Button component. This is to make sure users can easily interact with it. This is to comply with <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```

File: src/components/back-link/index.md [hunk 6]
```diff
@@ -52,12 +52,12 @@ Make sure the link takes users to the previous page they were on, in the state t
...
+<p>Do not pre-populate if the information is no longer valid, or when pre-populating would be a major safety or security concern. This is to comply with <a href="https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry.html">WCAG 2.2 success criterion 3.3.7 Redundant entry</a>.</p>
 {% endcall %}
 
 {% call wcagNote({id: "wcag-interact-back-links"}) %}
 
-<p>Make sure no other interactive elements are too close to the back link. This is to make sure users can easily interact with the back link. This relates to WCAG 2.2 success criterion <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">2.5.8 Target size (minimum)</a>.</p>
+<p>Make sure no other interactive elements are too close to the Back link component. This is to make sure users can easily interact with the Back link component. This relates to <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```
Comment (calvin-lau-sig7) [inline src/components/back-link/index.md:60]:
```suggestion
<p>Make sure no other interactive elements are too close to the Back link component. This is to make sure users can easily interact with it. This relates to <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```

File: src/components/breadcrumbs/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (calvin-lau-sig7) [reply src/components/breadcrumbs/index.md:48]:
```suggestion
<p>Make sure no other interactive elements are too close to the Breadcrumbs component. This is to make sure users can easily interact with it. This relates to <a href="https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html">WCAG 2.2 success criterion 2.5.8 Target size (minimum)</a>.</p>
```

**Review by owenatgov — COMMENTED** [review]
Content is looking SHARP. This is all but approved. I'll say my usual commit history thing. Happy to chat through that more.
